backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/bpartner/model/interceptor/C_BPartner.java

```diff
@@ -205,4 +209,26 @@ public void validateSalesRep(final I_C_BPartner partner)
 		final BPartnerId salesRepId = BPartnerId.ofRepoIdOrNull(partner.getC_BPartner_SalesRep_ID());
 		bPartnerBL.validateSalesRep(bPartnerId, salesRepId);
 	}
+
+	@ModelChange(timings = ModelValidator.TYPE_AFTER_NEW)
+	public void onNewPartner(final I_C_BPartner partner)
+	{
+		final BPartnerId bPartnerId = BPartnerId.ofRepoId(partner.getC_BPartner_ID());
+
+		Services.get(ITrxManager.class)
```

Comment:
It's advisable to put the services initializations at the beginning of the class so we have them all gathered together. This way, next time we need a service we won't have to initialize it again but we can reuse the class field. See, for example, de.metas.externalsystem.grssignum.export.interceptor.C_BPartner